### PR TITLE
[FEAT] CRPS, MSSE and Energy Score metrics

### PR DESCRIPTION
**Pull request by kdgutier**
General improvements to evaluation utilities
- Added `scaled_crps`, `msse` and `energy_score` metrics to evaluation.py
- Added running tests for metrics in `probabilistic_methods.ipynb`
- Added multivariate scoring and hierarchical accuracy references
- General improvements to the `evaluation.ipynb` documentation
- Changed default output samples.shape from [samples, N,H] to [N,H,samples], updated dependencies accordingly

PENDING
- Validate metrics using `core.HierarchicalReconciliation` in `TourismL` experiment.
- Update `core.HierarchicalReconciliation` to produce online samples or store memory through `fit`, `predict`/`sample`.

### CODE REVIEW

File: hierarchicalforecast/evaluation.py [hunk 1]
```diff
@@ -1,17 +1,231 @@
...
 
 import numpy as np
 import pandas as pd
 
-# %% ../nbs/evaluation.ipynb 5
-class HierarchicalEvaluation:    
+# %% ../nbs/evaluation.ipynb 6
+def _metric_protections(y: np.ndarray, y_hat: np.ndarray, 
+                        weights: Optional[np.ndarray]) -> None:
+    assert (weights is None) or (np.sum(weights) > 0), 'Sum of weights cannot be 0'
```
Comment (cchallu) [inline hierarchicalforecast/evaluation.py:16]: Use Exception instead of assert.

File: hierarchicalforecast/evaluation.py [hunk 2]
```diff
@@ -1,17 +1,231 @@
...
+    through a numerical integration technique, that discretizes the quantiles 
+    and treats the CRPS integral with a left Riemann approximation, averaging over 
+    uniformly distanced quantiles.    
+
+    $$ \mathrm{CRPS}(y_{\\tau}, \mathbf{\hat{F}}_{\\tau}) = \int^{1}_{0} \mathrm{QL}(y_{\\tau}, \hat{y}^{(q)}_{\\tau}) dq $$
+
+    **Parameters:**<br>
+    `y`: numpy array, Actual values.<br>
+    `y_hat`: numpy array, Predicted values.<br>
+    `quantiles`: numpy array,(n_quantiles). Quantiles to estimate from the distribution of y.<br>
```
Comment (cchallu) [inline hierarchicalforecast/evaluation.py:78]: Specify quantiles should be in [0,1] interval.

**Review by cchallu — APPROVED** [review]

File: hierarchicalforecast/evaluation.py (hunk 2 above)
Comment (kdgutier) [reply hierarchicalforecast/evaluation.py:78]: Done, added protection
And protection's unit test.

File: hierarchicalforecast/evaluation.py (hunk 1 above)
Comment (kdgutier) [reply hierarchicalforecast/evaluation.py:16]: Done

**Pull request merged**